### PR TITLE
MINIFICPP-1634 remove namespace aliases from Core.h

### PR DESCRIPTION
**Pull request by szaszm**
and adapt code that depends on them by using local aliases or
sufficiently qualified names.

The change started from an effort to move implementations from headers
to source files, so some unrelated refactoring is included, at least
until someone objects.

Long PR, but should be easy to review due to mostly repetitive trivial changes with localized effects only.

---

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE file?
- [x] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: libminifi/src/utils/crypto/EncryptionManager.cpp [hunk 1]
```diff
@@ -32,11 +32,7 @@ namespace minifi {
 namespace utils {
 namespace crypto {
 
-#ifdef WIN32
-constexpr const char* DEFAULT_NIFI_BOOTSTRAP_FILE = "\\conf\\bootstrap.conf";
```
Comment (fgerlits) [inline libminifi/src/utils/crypto/EncryptionManager.cpp:36]: Why did you change this?  If we want to change it, should DEFAULT_NIFI_PROPERTIES_FILE be changed similarly, too?  Either way, I don't think this change should be part of this pull request.

File: libminifi/include/utils/file/FileManager.h [hunk 2]
```diff
@@ -17,18 +17,10 @@
 #ifndef LIBMINIFI_INCLUDE_UTILS_FILE_FILEMANAGER_H_
 #define LIBMINIFI_INCLUDE_UTILS_FILE_FILEMANAGER_H_
 
+#include <filesystem>
```
Comment (fgerlits) [inline libminifi/include/utils/file/FileManager.h:20]: I think these `std::filesystem`-related changes should be in a separate pull request.

**Review by fgerlits — COMMENTED** [review]
Full disclosure: I mostly just scrolled through this, and only looked more closely at a few files selected at random, but with this method, the changes looked good. :)

Thanks for the cleanup!

File: libminifi/src/utils/crypto/EncryptionManager.cpp (hunk 1 above)
Comment (szaszm) [reply libminifi/src/utils/crypto/EncryptionManager.cpp:36]: This was because I changed concat_path to use std::filesystem::path /=, and concatenating "\conf\bootstrap.conf" (an absolute path) to anything results in this second part overwriting the path of the first. This is indeed a behavior change that would ideally not be part of this PR. The reason it is is that I started with this change in the same working directory and I wanted to avoid separating the changes if I could get away leaving them as they are.

Comment (szaszm) [reply libminifi/src/utils/crypto/EncryptionManager.cpp:36]: I would still prefer not separating them, so unless you insist, I would prefer to have these two independent changes reviewed together. Of course this only works if all changes are acceptable.

Comment (fgerlits) [reply libminifi/src/utils/crypto/EncryptionManager.cpp:36]: I definitely don't want `DEFAULT_NIFI_PROPERTIES_FILE == "\\conf\\minifi.properties"` and `DEFAULT_NIFI_BOOTSTRAP_FILE == "./conf/bootstrap.conf"` on Windows, so if one of them changes, the other should, too.

I would prefer this change to be part of a separate, `std::filesystem`-related PR, but if it is too much work to separate them, I am going to approve it is at is (after the properties file change).

File: extensions/standard-processors/processors/RouteOnAttribute.h [hunk 3]
```diff
@@ -39,9 +39,8 @@ namespace processors {
 
 class RouteOnAttribute : public core::Processor {
  public:
-  RouteOnAttribute(const std::string& name, const utils::Identifier& uuid = {}) // NOLINT
-      : core::Processor(name, uuid),
-        logger_(logging::LoggerFactory<RouteOnAttribute>::getLogger()) {
+  explicit RouteOnAttribute(const std::string& name, const utils::Identifier& uuid = {}) // NOLINT
```
[the remark this answers is not in the document]
Comment (szaszm) [reply extensions/standard-processors/processors/RouteOnAttribute.h]: It's no longer needed, I've removed it during the latest rebase

File: libminifi/src/utils/crypto/EncryptionManager.cpp (hunk 1 above)
Comment (szaszm) [reply libminifi/src/utils/crypto/EncryptionManager.cpp]: I've removed all filesystem-related changes in the latest rebase
